src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -492,104 +469,33 @@ public QueueSet getQueueSet(TaskPriority priority) {
-      runningThread.start();
-    }
-
-    @Override
-    protected void shutdownService() {
-      Thread runningThread = this.runningThread;
-      this.runningThread = null;
-      runningThread.interrupt();
-    }
-
-    @Override
-    public void run() {
-      Worker worker = null;
-      while (runningThread != null) {
-        try {
-          worker = workerPool.getWorker();
-          if (worker != null) {
-            TaskWrapper nextTask = getNextReadyTask();
-            if (nextTask != null) {
-              worker.nextTask(nextTask);
-            } else {
-              workerPool.workerDone(worker);
-            }
-          }
-        } catch (InterruptedException e) {
-          if (worker != null) {
-            workerPool.workerDone(worker);
-          }
-          stopIfRunning();
-          Thread.currentThread().interrupt();
-        } catch (Throwable t) {
-          if (worker != null) {
-            // we kill the worker here because this condition is not expected
-            workerPool.killWorker(worker);
-          }
-          ExceptionUtils.handleException(t);
-        } finally {
-          worker = null;
-        }
-      }
-    }
     
     /**
-     * Gets the next task that is ready to execute immediately.  This call gets the next high and 
-     * low priority task, compares them and determines which one should be executed next.
+     * Gets the next task currently queued for execution.  This task may be ready to execute, or 
+     * just queued.  If a queue update comes in, this must be re-invoked to see what task is now 
+     * next.  If there are no tasks ready to be executed this will simply return {@code null}.
      * 
-     * If there are no tasks ready to be executed this will block until one becomes available 
-     * and ready to execute.
-     * 
-     * @return Task to be executed next, or {@code null} if shutdown while waiting for task
-     * @throws InterruptedException Thrown if calling thread is interrupted while waiting for a task
+     * @return Task to be executed next, or {@code null} if no tasks at all are queued
      */
-    protected TaskWrapper getNextReadyTask() throws InterruptedException {
-      while (runningThread != null) {  // loop till we have something to return
-        TaskWrapper nextTask = getNextTask(highPriorityQueueSet, lowPriorityQueueSet, 
-                                           maxWaitForLowPriorityInMs);
-        if (nextTask == null) {
+    public TaskWrapper getNextTask() {
+      TaskWrapper nextTask = AbstractPriorityScheduler.getNextTask(highPriorityQueueSet, 
+                                                                   lowPriorityQueueSet, 
+                                                                   maxWaitForLowPriorityInMs);
+      if (nextTask == null) {
+        return starvablePriorityQueueSet.getNextTask();
+      } else {
+        long nextTaskDelay = nextTask.getScheduleDelay();
+        if (nextTaskDelay > 0) {
```

Comment:
Have you tried setting this to > 1 or maybe even up to 5?  I dont know that its worth doing any other logic if we almost need to run it so soon.